### PR TITLE
Fix #852 - Process Qute templates in a strict manner

### LINKED ISSUE #852: Cannot find symbol "NOT_FOUND.class"
During generation with 2.6.0-lts (or 2.5.0) from an Open API JSON file, a NOT_FOUND.class is included in annotations but **not imported** :

```java
@com.fasterxml.jackson.annotation.JsonSubTypes({
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = NOT_FOUND.class, name = "ArchivalAutoFillExpansion"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = NOT_FOUND.class, name = "ReindexAutoFillExpansion"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = NOT_FOUND.class, name = "ArchivalAutoFillExpansionModifierProperties"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = NOT_FOUND.class, name = "ReindexAutoFillExpansionModifierProperties"),
})
```

It was working with 2.4.7.

### PR DESCRIPTION
**Pull request by ricardozanini**
Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

In this PR:
- Removed XML support on template code generation. It was never tested, possibly unstable.
- Turn Qute Strict processing, see more: https://quarkus.io/guides/qute-reference#strict_rendering
- Completely review of *.qute templates, variables, processing

I'm surprised a few of these features were working - at all—with many parsing logic and errors. 

Fix #852 

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-quarkus2` to backport the original PR to the `quarkus2` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by ricardozanini** [comment]
@david-cabillic can you take a look and try it on your end?

**Comment by ricardozanini** [comment]
@mcruzdev @hbelmiro please carefully review it.

**Comment by david-cabillic** [comment]
@ricardozanini I have a Maven build issue:

 [ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.16.4:generate-code (default) on project quarkus-document-api-client: Quarkus code generation phase has failed: InvocationTargetException: Could not generate model 'PostIndexingModifier_request': TemplateException: Rendering error: Property "classname" not found on the base object "null" in expression {child.model.classname} -> [Help 1]

**Comment by ricardozanini** [comment]
Are you using the same openapi you sent to us in the issue? Because this file is here in the PR and I can't see this error. 

If no, please send me the file.

**Comment by david-cabillic** [comment]
@ricardozanini I have 2 files in my project.

**Review by fjtirado — APPROVED** [review]

**Comment by ricardozanini** [comment]
> @ricardozanini I have 2 files in my project.

Can you please share?

**Comment by ricardozanini** [comment]
@david-cabillic I figured it out.

Given this openapi excerpt:

```yaml
    CheckpointStatistics:
      required:
        - className
      type: object
      properties:
        className:
          type: string
      discriminator:
        propertyName: className
        mapping:
          completed: '#/components/schemas/CompletedCheckpointStatistics'
          failed: '#/components/schemas/FailedCheckpointStatistics'
          in_progress: '#/components/schemas/PendingCheckpointStatistics'
```

It will generate annotations like this:

```java
@com.fasterxml.jackson.annotation.JsonIgnoreProperties(
  value = "className", // ignore manually set className, it will be automatically generated by Jackson during serialization
  allowSetters = true // allows the className to be set during deserialization
)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use = com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include = com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property = "className", visible = true)
@com.fasterxml.jackson.annotation.JsonSubTypes({
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = CompletedCheckpointStatistics.class, name = "completed"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = FailedCheckpointStatistics.class, name = "failed"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = PendingCheckpointStatistics.class, name = "in_progress"),
})
public class CheckpointStatistics  { }
```

It wasn't working - at all - in previous versions.

Can you try again my new commit? Please make sure that your `discriminator` field in your spec file is correct.

**Comment by ricardozanini** [comment]
Actually I need to fix the import first.

**Comment by david-cabillic** [comment]
[administration.json](https://github.com/user-attachments/files/17871880/administration.json)
[document.json](https://github.com/user-attachments/files/17871882/document.json)
This 2 documents I'm using.

`properties
# https://docs.quarkiverse.io/quarkus-openapi-generator/dev/client.html
quarkus.openapi-generator.codegen.enable-security-generation=false
# administration -> https://sdk.onbase.com/rest/docs.html?product=OnBase&release=Foundation24.1&spec=onbase-administration
quarkus.openapi-generator.codegen.spec.administration_json.base-package=ch.tcs.onbase.administration
quarkus.openapi-generator.codegen.spec.administration_json.config-key=obadministration
quarkus.openapi-generator.codegen.spec.administration_json.additional-api-type-annotations=@org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders(ch.tcs.onbase.header.AuthHeadersFactory.class)
quarkus.openapi-generator.codegen.spec.administration_json.model-name-prefix=OB
# document -> https://sdk.onbase.com/rest/docs.html?product=OnBase&release=Foundation24.1&spec=document
quarkus.openapi-generator.codegen.spec.document_json.base-package=ch.tcs.onbase.document
quarkus.openapi-generator.codegen.spec.document_json.config-key=obdocument
quarkus.openapi-generator.codegen.spec.document_json.additional-api-type-annotations=@org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders(ch.tcs.onbase.header.AuthHeadersFactory.class)
quarkus.openapi-generator.codegen.spec.document_json.model-name-prefix=OB
quarkus.openapi-generator.codegen.spec.document_json.type-mappings.File=byte[]`

**Comment by ricardozanini** [comment]
@david-cabillic can you please update and try again?

**Comment by david-cabillic** [comment]
@ricardozanini Build ok, well done .

**Comment by ricardozanini** [comment]
@david-cabillic, as soon as @mcruzdev and @hbelmiro review this PR, I'll merge it and release a new minor version. Thanks for reporting it; I've found many problems in the templates thanks to it.

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapper.java [hunk 1]
```diff
@@ -78,9 +75,46 @@ public abstract class OpenApiClientGeneratorWrapper {
...
+        this.configurator.addAdditionalProperty("skipFormModel", TRUE);
+        this.configurator.addAdditionalProperty("templateDir", "");
+        this.configurator.addAdditionalProperty("use-bean-validation", FALSE);
+        this.configurator.addAdditionalProperty("use-field-name-in-part-filename", FALSE);
+        this.configurator.addAdditionalProperty("verbose", FALSE);
+        // TODO: expose as properties
+        this.configurator.addAdditionalProperty(CodegenConstants.SERIALIZABLE_MODEL, FALSE);
+    }
+
     public OpenApiClientGeneratorWrapper withApiPackage(final String pkg) {
```
Comment (hbelmiro) [inline client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapper.java:118]: `withApiPackage` and `withModelPackage` seem no longer used. Can we remove them?

**Review by hbelmiro — APPROVED** [review]
LGTM
I just left two optional comments.

**Comment by david-cabillic** [comment]
@ricardozanini You're welcome

**Pull request merged**